quill-core/src/main/scala/io/getquill/context/Context.scala

```diff
@@ -7,7 +7,7 @@ import java.io.Closeable
 import scala.util.Try
 import io.getquill.NamingStrategy
 
-trait Context[Idiom <: io.getquill.idiom.Idiom, Naming <: NamingStrategy]
+abstract class Context[Idiom <: io.getquill.idiom.Idiom, Naming <: NamingStrategy](val idiom: Idiom, val naming: Naming)
```

Comment:
What's the purpose to make it `abstract class`? abstract public members `idiom` and `naming` won't work?

Reply:
I've just tried to make it a trait again. I didn't like the result, let me know if you feel strongly about it.

Reply:
nevermind, I had to change it back to a trait because of a scala.js bug

Reply:
Sorry for late response, it is ok for me both cases, I was just curious